vlib/v/checker/tests/index_sumtype_or_interface_params_err.vv

```diff
@@ -0,0 +1,8 @@
+type Foo = int | map[int]int
+
+interface Bar {}
+
+fn sum_type_or_interface_as_parameters(mut foo Foo, mut bar Bar) {
```

Comment:
Please also test for `struct MyStruct {}`, then `mut zz MyStruct` and `zz[0]` - this should be an error.

Also `yy &MyStruct`, then `zz[0]`, should not be.

Reply:
Ok, I'll add a test for `Struct`, It just so happens that the next PR is add `mut zz MyStruct` and `zz[0]` need `unsafe block` notification, can be submitted together, thanks for reminder.  ❤️

Reply:
done, https://github.com/vlang/v/pull/19992  :)